### PR TITLE
Mnot 958

### LINKED ISSUE #958: Selected [representation, response]
We use the phrases 'selected representation' and 'selected response' in Semantics and Caching (respectively) to refer to two completely different concepts.

Can we do better?

### PR DESCRIPTION
**Pull request by mnot**
Additional instances of 'select'

Fixes #958

### CODE REVIEW

**Comment by mnot** [comment]
I went with 'nominated' in a few places where 'choosing' didn't seem workable. 'varying' could work too, I suppose, but we'd probably need to add it as a defined term.

**Comment by royfielding** [comment]
I think that "selecting header fields" was fine and reads better than nominated or chosen fields — they are the request fields that the origin claims were used to select each representation, so the name fits even if we use them later to choose a response.

File: draft-ietf-httpbis-cache-latest.xml [hunk 1]
```diff
@@ -582,12 +582,12 @@
...
    the cache &MUST-NOT; use that stored response without revalidation unless
-   all the selecting header fields (nominated by that Vary field value) in the
-   present request match those fields in the original request (i.e., the
+   all the presented request header fields nominated by that Vary field value
+   match those fields in the original request (i.e., the
    request that caused the cached response to be stored).
 </t>
 <t>
-   The selecting header fields from two requests are defined to match if
+   The nominated header fields from two requests are defined to match if
```
Comment (mnot) [inline draft-ietf-httpbis-cache-latest.xml:590]:
```suggestion
   The header fields from two requests are defined to match if
```

File: draft-ietf-httpbis-cache-latest.xml [hunk 2]
```diff
@@ -616,33 +616,34 @@
...
-   <x:dfn>selected response</x:dfn>.
-</t>
-<t>
-   If multiple selected responses are available (potentially including
-   responses without a Vary header field), the cache will need to choose one to use.
-   When a selecting header field has a known mechanism for doing so (e.g., qvalues on
-   <x:ref>Accept</x:ref> and similar request header fields), that mechanism &MAY; be
-   used to select a preferred response. If such a mechanism is not available, or leads to equally preferred responses, the most recent
-   response (as determined by the <x:ref>Date</x:ref> header field) is used, as
+   If multiple stored responses match the nominated header fields
```
Comment (mnot) [inline draft-ietf-httpbis-cache-latest.xml:620]:
```suggestion
   If multiple stored responses match the request header fields nominated by Vary
```

**Comment by mnot** [comment]
@royfielding see updates; I think I managed to avoid the awkwardness

File: draft-ietf-httpbis-cache-latest.xml [hunk 3]
```diff
@@ -278,7 +278,7 @@
 <t>
    If a request target is subject to content negotiation, the cache might
    store multiple responses for it. Caches differentiate these responses
-   by incorporating values of the original request's selecting header fields
+   by incorporating chosen values from the original request's header fields
```
Comment (royfielding) [inline draft-ietf-httpbis-cache-latest.xml:281]:
```suggestion
   by incorporating some of the original request's header fields
```

File: draft-ietf-httpbis-cache-latest.xml [hunk 4]
```diff
@@ -616,33 +616,34 @@
...
-</t>
-<t>
-   If multiple selected responses are available (potentially including
-   responses without a Vary header field), the cache will need to choose one to use.
-   When a selecting header field has a known mechanism for doing so (e.g., qvalues on
-   <x:ref>Accept</x:ref> and similar request header fields), that mechanism &MAY; be
-   used to select a preferred response. If such a mechanism is not available, or leads to equally preferred responses, the most recent
-   response (as determined by the <x:ref>Date</x:ref> header field) is used, as
+   If multiple stored responses match the request header fields nominated by Vary
+   (potentially including responses without a Vary header field),
```
Comment (royfielding) [inline draft-ietf-httpbis-cache-latest.xml:621]:
```suggestion
   If multiple stored responses match,
```

File: draft-ietf-httpbis-cache-latest.xml [hunk 5]
```diff
@@ -616,33 +616,34 @@
...
-   more preferable responses are available. When a cache has multiple responses for a
-   target URI and one or more omits the Vary header field, it &SHOULD; use the
-   most recent (see <xref target="age.calculations"/>) valid Vary field value available to select an appropriate response
-   for the request.
+   response (i.e., the one sent when the request does not express any preferences),
+   with the effect of choosing it for subsequent requests to that resource
+   even when more preferable responses are available. When a cache has
+   multiple stored responses for a target URI and one or more omits the Vary
+   header field, the cache &SHOULD; choose the most recent
+   see <xref target="age.calculations"/>) stored response with a valid Vary
```
Comment (royfielding) [inline draft-ietf-httpbis-cache-latest.xml:639]:
```suggestion
   header field, the cache &SHOULD; choose the most recent
   (see <xref target="age.calculations"/>) stored response with a valid Vary
```

File: draft-ietf-httpbis-cache-latest.xml (hunk 3 above)
Comment (royfielding) [reply draft-ietf-httpbis-cache-latest.xml:281]: Also, the first sentence would be better as

>   A cache might store multiple responses for a request target that is subject to content negotiation.

**Review by royfielding — CHANGES_REQUESTED** [review]

File: draft-ietf-httpbis-cache-latest.xml [hunk 6]
```diff
@@ -278,7 +278,7 @@
 <t>
    If a request target is subject to content negotiation, the cache might
    store multiple responses for it. Caches differentiate these responses
```
Comment (mnot) [inline draft-ietf-httpbis-cache-latest.xml:280]:
```suggestion
   A cache might store multiple responses for a request target that is subject 
   to content negotiation. Caches differentiate these responses
```

**Review by royfielding — APPROVED** [review]

**Review by mosi2021v — COMMENTED** [review]
#

**Pull request merged**